static/Functions.js

```diff
@@ -131,30 +130,43 @@ function storeAttachments(event) {
     console.log('got response', response);
 
     var trackingIds = [];
+    var errors = [];
+    var proofsToStore = [];
     if (response.attachmentProcessingDetails) {
       for (i = 0; i < response.attachmentProcessingDetails.length; i++) {
-
         var ad = response.attachmentProcessingDetails[i];
-        var proof = {
+        proofsToStore.push({
           proofToEncrypt: {
             blobName: ad.blobName,
             documentName: ad.name
           },
           publicProof: {
             documentHash: ad.hash
           }
-        };
-
-        return putProof(proof, function(err, response) {
-          if (err) return showMessage(err.message, event);
-
-          trackingIds.push(response.trackingId);
-
-          Office.context.mailbox.item.displayReplyForm(JSON.stringify(trackingIds));
-          return showMessage("Attachments processed: " + JSON.stringify(trackingIds), event);
         });
       }
     }
+
+    return putProofs(proofsToStore, function(err, response) {
```

Comment:
this should be within the previous 'if' block...

Reply:
Done, and added "else" block